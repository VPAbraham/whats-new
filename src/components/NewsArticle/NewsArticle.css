.NewsArticle {
  width: 45%;
  margin: auto;
  background: rgba(0, 47, 128, 0.187); 
}

.NewsArticle-img {
  width: 100%;
  border-radius: 5px;
  border: 3px solid rgba(51, 6, 6, 0.123);
}

.NewsArticle-headline {

}

.NewsArticle-description {

}

.NewsArticle-link {

}