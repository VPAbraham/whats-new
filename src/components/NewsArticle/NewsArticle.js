import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  return (
    <div className="NewsArticle">
      <img src={props.img} className="NewsArticle-img"  alt="news article header"/>
      <h3 className="NewsArticle-headline">{props.headline}</h3>
      <p className="NewsArticle-description">{props.description}</p>
      <a href={props.link} className="NewsArticle-link">Click for Link</a>
    </div>
  )
}

export default NewsArticle;