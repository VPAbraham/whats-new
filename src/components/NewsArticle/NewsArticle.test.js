import React from 'react'
import { shallow } from 'enzyme'
import NewsArticle from './NewsArticle'

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<div className="NewsArticle">
      <img className="NewsArticle-img" src={props.img} alt="news article header" />
      <h3 className="NewsArticle-headline">{props.headline}</h3>
      <p className="NewsArticle-description">{props.description}</p>
      <a className="NewsArticle-link" href={props.url}>Click for Link</a>
    </div>)

    expect(wrapper).toMatchSnapshot()
  })
})