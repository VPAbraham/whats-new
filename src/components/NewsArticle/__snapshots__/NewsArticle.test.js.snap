// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<div
  className="NewsArticle"
>
  <img
    alt="news article header"
    className="NewsArticle-img"
  />
  <h3
    className="NewsArticle-headline"
  />
  <p
    className="NewsArticle-description"
  />
  <a
    className="NewsArticle-link"
  >
    Click for Link
  </a>
</div>
`;
